app/models/adviser.rb

```diff
@@ -10,9 +10,9 @@ class Adviser < ActiveRecord::Base
   has_and_belongs_to_many :professional_standings
   has_and_belongs_to_many :professional_bodies
 
-  before_validation :assign_name, if: :reference_number?
-
   before_validation :upcase_postcode
+  before_validation :upcase_reference_number
```

Comment:
This will basically affect the way every new record is also inserted, which is actually good. Like it 👍